core/trino-main/src/main/java/io/trino/execution/BasicStageStats.java

```diff
@@ -72,9 +74,11 @@
     private final DataSize rawInputDataSize;
     private final long rawInputPositions;
     private final long cumulativeUserMemory;
+    private final long failedCumulativeUserMemory;
     private final DataSize userMemoryReservation;
     private final DataSize totalMemoryReservation;
     private final Duration totalCpuTime;
+    private final Duration failedCpuTime;
```

Comment:
maybe "wasted" (here and in other places)?

Reply:
I personally like `failed` more as it is more precise and does not add emotional meaning to json field. But I do not feel super strongly. @martint / @findepi opinion?

Reply:
i'd prefer "failed" too.

btw do  you plan anything like speculative execution, where work could become "wasted" without being "failed"?

Reply:
eg "cancelled"

Reply:
Yeah - that will come up. But even then "wasted" is wrong term really. It has negative connotation, and if we are up to kill some tasks to speed up the execution of whole query it should not be seen as we are doing something wrong. And "wasting" resources feels wrong. So maybe some other term - neither "failed" nor "wasted" - but nothing comes to my mind.

Reply:
FWIW, I think "failed" is ok. The failure could be intrinsic (cluster / resource / system issues) or induced (task was forced to abort). In either case, it failed to complete.